packages/components/pie-radio/src/index.ts

```diff
@@ -12,9 +15,50 @@ const componentSelector = 'pie-radio';
 /**
  * @tagname pie-radio
  */
-export class PieRadio extends RtlMixin(LitElement) implements RadioProps {
+export class PieRadio extends FormControlMixin(RtlMixin(LitElement)) implements RadioProps {
+    @property({ type: Boolean })
+    public checked = defaultProps.checked;
+
+    @property({ type: Boolean })
+    public defaultChecked = defaultProps.defaultChecked;
+
+    @property({ type: Boolean })
+    public disabled = defaultProps.disabled;
+
+    @property({ type: String })
+    public name: RadioProps['name'];
+
+    @property({ type: Boolean })
+    public required = defaultProps.required;
+
+    @property({ type: String })
+    public value = defaultProps.value;
+
+    @query('input[type="radio"]')
+    private radio!: HTMLInputElement;
+
     render () {
-        return html`<h1 data-test-id="pie-radio">Hello world!</h1>`;
+        const {
+            checked, disabled, name, required, value,
+        } = this;
+
+        return html`
+        <input
+            type="radio"
+            id="radioId"
+            data-test-id="pie-radio"
+            .checked="${live(checked)}"
+            .value="${value}"
+            name="${ifDefined(name)}"
+            ?disabled="${disabled}"
+            ?required="${required}">
+        <label for="radioId">
+            <slot></slot>
+        </label>`;
+    }
+
+    public get validity () : ValidityState {
+        return this.radio.validity;
     }
```

Comment:
I am not sure how strictly we follow the files structure but for other components `render()`  part is the last one, so can we move this getter higher up?

Reply:
Interesting point. I would like it if we could have some linting rules about ordering of methods and properties. I remember Vue being quite opinionated with this sort of thing.